src/composeappmanager.cc

```diff
@@ -210,8 +212,12 @@ DownloadResult ComposeAppManager::Download(const TufTarget& target) {
 
   for (const auto& pair : all_apps_to_fetch) {
     LOG_INFO << "Fetching " << pair.first << " -> " << pair.second;
-    if (!app_engine_->fetch({pair.first, pair.second})) {
-      res = {DownloadResult::Status::DownloadFailed, "Download failed TODO"};
+    const auto fetch_res{app_engine_->fetch({pair.first, pair.second})};
+    if (!fetch_res) {
+      const std::string errDesc{boost::str(boost::format("failed to fetch App; app: %s; uri: %s; err: %s") %
+                                           pair.first % pair.second % fetch_res.err)};
+      res = {DownloadResult::Status::DownloadFailed, errDesc};
+      // Do we want to continue with Apps fetching after a failure to fetch one of the Apps?
```

Comment:
probably isn't a bad idea. one problem with this logic - if there are >1 errors, you overwrite the previous `res.message`

Reply:
yes, we can concatenate error messages. I am just not sure it's worth trying to fetch other Apps if one of them fails, especially if we consider Target as atomic.

Reply:
that's fine be me. odds are if the first one fails, the 2nd one will also.

Reply:
just added `break` if the app fetch fails.